### PR TITLE
Update dependency go modules for k8s v1.28.0-rc.0

### PR DESCRIPTION
**Pull request by Sneha-at**
Ran kubernetes-csi/release-tools go-get-kubernetes.sh -p 1.28.0-rc.0

```release-note
NONE
```

### CODE REVIEW

**Comment by mauriciopoppe** [comment]
I think you synced release-tools at the same time as bumping a library, could you please split the PRs?

**Review by mauriciopoppe — COMMENTED** [review]
I understand and it's already in 2 commits so it's good

/lgtm
/approve

**Comment by mauriciopoppe** [comment]
/assign @msau42 @jsafrane 
For approval

File: go.mod [hunk 1]
```diff
@@ -34,18 +34,27 @@ require (
...
 	gopkg.in/yaml.v3 v3.0.1 // indirect
-	k8s.io/apimachinery v0.27.4 // indirect
-	k8s.io/component-base v0.27.4 // indirect
-	k8s.io/kube-openapi v0.0.0-20230501164219-8b0f38b5fd1f // indirect
+	k8s.io/apimachinery v0.28.0-rc.0 // indirect
+	k8s.io/component-base v0.28.0-rc.0 // indirect
+	k8s.io/kube-openapi v0.0.0-20230717233707-2695361300d9 // indirect
 )
+
+replace k8s.io/api => k8s.io/api v0.28.0-rc.0
```
Comment (sunnylovestiramisu) [inline go.mod:52]: Remove the replace or if you can find out why the script is doing this.

**Comment by msau42** [comment]
There's a few more changes in release tools we should pick up. Can you refresh the release-tools update after https://github.com/kubernetes-csi/csi-release-tools/pull/229 merges?

**Pull request merged**